### PR TITLE
Built zone / area inherit rework to use "unconnected zone"

### PR DESCRIPTION
**Pull request by Kubius**
[REWORK][INTERNAL]
## About the PR 

Overhauls the behavior of turf area allocation to use a new global unconnected zone.

When a turf attempts to inherit area, instead of only polling adjacent areas, it will be added to the unconnected zone, a global area that holds constructed turfs that don't belong to any particular "real" area, and then prompt the unconnected zone to evaluate it for area propagation using propagate_zone().

The unconnected zone, through propagate_zone(), handles propagation as follows.

- First: the turf being evaluated is marked as undergoing transfer evaluation, so other propagation doesn't attempt to work on the turf at the same time. A lagcheck is included immediately thereafter, as the propagation process isn't rate-limited otherwise.
- Second: check for the presence of an APC in the given turf. If there is one (either because an APC was built in the unconnected zone, or an unconnected zone was built onto an APC), a new Built Zone likely needs to be prepared for it, and the proc makes note of this.
- Third: evaluate adjacent turfs in cardinal directions. If there's at least one adjacent expandable area, the first one found is selected - adjacent turfs belonging to the unconnected zone are also enumerated in this phase, and placed in the "propagation_targets" list.
- Fourth: if there is a suitable recipient of the turf being evaluated (either an adjacent expandable area or a to-be-created Built Zone for a found APC), the turf will be transferred to that recipient using transfer_ownership, and propagate_zone() is then called targeting adjacent unconnected zone turfs in order to fully propagate any found area. If there is no suitable recipient, the turf will remain in the unconnected zone.
- Lastly, the turf is marked as no longer undergoing transfer evaluation, so further attempts at propagation will work.

APCs now prompt a zone propagation when they are constructed, and have a variable that can be poked by transfer_ownership to instruct them to refresh their member tiles the next time they process (avoids refreshing once for each newly-transferred tile).

*Before APC installation.*
image

*After APC installation.*
image

*Worst case scenario under current built zone system.*
image

## Why's this needed? 

Current built-zone behavior is unresponsive, unintuitive and can create a significant amount of tiny new areas that never get cleaned up. With this change, all disconnected built tiles use a single shared area so there are no new areas being created until an APC installation prompts it, connecting new construction to the station works consistently, and space construction without a room designator works reliably.

Changelog entry included as this is very relevant to construction enjoyers.

## Changelog 

```changelog
(u)Kubius
(+)Tiles constructed outside of an area will now use a single "unconnected zone" area by default, making it much easier to connect these areas to station areas or turn them into a single built zone using an APC.
```

### CODE REVIEW

**Comment by klookiyam** [comment]
Is it possible to build a distinct new zone adjacent to a current zone with this change? Would placing new adjacent tiles after an apc is placed automatically attempt to attach to the previously built zone?

**Comment by Kubius** [comment]
The room designator remains able to subdivide built zones, and you could also build an APC on a separate tile and extend your separate room outwards from that one.

**Review by TobleroneSwordfish — APPROVED** [review]
Code looks generally okay, suggested behaviour is more sane than current *I think*

File: code/modules/power/apc.dm [hunk 1]
```diff
@@ -75,6 +75,7 @@ ADMIN_INTERACT_PROCS(/obj/machinery/power/apc, proc/toggle_operating, proc/zapSt
 	var/timeout = 60 //The time until we auto disconnect (if we don't get a refresh ping)
 	var/timeout_alert = 0 //Have we sent a timeout refresh alert?
 	var/hardened = 0 // azone/listening post apcs that you dont want fucked with. immune to explosions, blobs, meteors
+	var/update_requested = FALSE
```
[the remark this answers is not in the document]
Comment (Kubius) [reply code/modules/power/apc.dm]: Huh. Had that commented at one point, not sure where it went. Fixed either way.

**Pull request merged**